marconi-chain-index/src/Marconi/ChainIndex/Indexers/EpochStakepoolSize.hs

```diff
@@ -1,187 +1,442 @@
-{-# OPTIONS_GHC -Wno-orphans      #-}
-{-# OPTIONS_GHC -Wno-overlapping-patterns #-}
-{-# LANGUAGE DataKinds         #-}
-{-# LANGUAGE FlexibleInstances #-}
-{-# LANGUAGE GADTs             #-}
-{-# LANGUAGE LambdaCase        #-}
-{-# LANGUAGE MultiWayIf        #-}
-{-# LANGUAGE OverloadedStrings #-}
-{-# LANGUAGE TypeApplications  #-}
-
-module Marconi.ChainIndex.Indexers.EpochStakepoolSize where
-
-import Control.Monad.Trans.Class (lift)
+{-# LANGUAGE DataKinds          #-}
+{-# LANGUAGE DerivingStrategies #-}
+{-# LANGUAGE FlexibleInstances  #-}
+{-# LANGUAGE GADTs              #-}
+{-# LANGUAGE NamedFieldPuns     #-}
+{-# LANGUAGE OverloadedStrings  #-}
+{-# LANGUAGE QuasiQuotes        #-}
+{-# LANGUAGE TemplateHaskell    #-}
+{-# LANGUAGE TupleSections      #-}
+
+-- | Module for indexing the stakepool delegation per epoch in the Cardano blockchain.
+--
+-- This module will create the SQL tables:
+--
+-- + table: epoch_spd
+--
+-- @
+--    |---------+--------+----------+--------+-----------------+---------|
+--    | epochNo | poolId | lovelace | slotNo | blockHeaderHash | blockNo |
+--    |---------+--------+----------+--------+-----------------+---------|
+-- @
+--
+-- To create this table, we need to compute the `LedgerState` from `ouroboros-network` (called
+-- `NewEpochState` in `cardano-ledger`) at each 'Rollforward' chain sync event. Using the
+-- 'LegderState', we can easily compute the epochNo as well as the stake pool delegation for that
+-- epoch.
+--
+-- The main issue with this indexer is that building the LedgerState and saving it on disk for being
+-- able to resume is VERY resource intensive. Syncing time for this indexer is over 20h and uses
+-- about ~16GB of RAM (which will keep increasing as the blockchain continues to grow).
+--
+-- Here is a synopsis of what this indexer does.
+--
+-- We assume a few things:
```

Comment:
I agree for the 2nd point. However, the first _is_ an assumption. I'm assuming that the LedgerState is contructed _outside_ the module.

Reply:
One assumes something that they don't know to be true (but assume it to be true), so this would mean that we don't know that the code is in a different file. But I think we do know where the code is.

Reply:
Ahh :D Got it. So I should be talking about a precondition instead?

Reply:
Why mention it at all? The actual implementation of applying blocks to ledger state is in the ouroboros repo anyway and all of us, including marconi, cardano-node, cardano-api and perhaps others import it from there.

Reply:
Why not keep indexer code inside the same module then? Not only do you keep code for a single indexer in a single module you can also skip this comment.